src/api_rework.jl

```diff
@@ -0,0 +1,194 @@
+# I am putting everything in here for now. The contents of this file should be moved around in the future.
+
+"""
+    peakproms!(pks) --> NamedTuple
+    peakproms!() --> Function
```

Comment:
I disagree; I think that putting all the different methods docs in the same docstring makes it harder to find the differences in use/etc between the different methods, and makes the description longer. I prefer this:

```
    function(x, y, z) -> a
Complete description
Examples
───────────
    method(y, z)
Specific, shorter description of how this method differs from main
Maybe examples
...
```

to this:

```
    function(x, y, z) -> a
Complete description
Clarify method1...
Clarify method2...
etc
Examples: main function
examples: method1
etc
...
```

Specifically, the kind of "structural" formatting that separates docstrings helps guide the eye when a user knows they are looking for documentation of one of the docstrings (eg a curried method)

Reply:
I think that it is actually really difficult to make accessible docstrings (not overwhelming) in a good way here. I really think that we need to concider using the `# Extended help` header (explained [here](https://docs.julialang.org/en/v1/manual/documentation/)). Information like
> `pks` must have `:indices` and `:heights` fields, at a minimum.
The following fields will also be copied and filtered if they exist: `:proms`, `:widths`,
and `:edges`.

belongs in there IMO, because it will generally just work, so most users never need to think about or understand this fact. There really is a "less is more" effect at play here, where documenting every aspect at once makes me understand _less_, when I am in fact just looking for valid signatures. Would you be open to making doctrings more minimal, adding the rest of the detail in the extended help?

Reply:
I just updated the doctring for `peaksheights` to use separate examples, because I found that trying to do both ended up being more natural for one API than the other, and being somewhat contrived. Potentially just more confusing even, in terms of showing how one should use the function. How do you feel about that?

If we manage to slim down the docstrings, and combine it into a single one, then I think we could have a very short seperate doctring for the curried versions. That actually sounds pretty great! Just saying it uses the same arguments, and then one example.